File: apps/mo2-android/android/core/model/src/main/kotlin/jp/developershub/dub/mo2/core/model/Notification.kt
// Mirror of @dub/types `notification` namespace (packages/types/src/notification.ts).
package jp.developershub.dub.mo2.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class NotificationChannel {
    @SerialName("in_app") IN_APP,
    @SerialName("email") EMAIL,
    @SerialName("chat") CHAT,
    @SerialName("push") PUSH,
}

// NotificationType is an open vocabulary (string) in the frozen contract.
typealias NotificationType = String

@Serializable
data class InboxItem(
    val id: NotificationId,
    val type: NotificationType,
    val title: String,
    val body: String,
    val readAt: ISODateTime? = null,
    val createdAt: ISODateTime,
    val resourceType: String? = null,
    val resourceId: String? = null,
)

// ListInboxResponse = Paginated<InboxItem> (frozen alias).
typealias ListInboxResponse = Paginated<InboxItem>

@Serializable
data class PreferenceEntry(
    val type: NotificationType,
    val channels: List<NotificationChannel>,
)

File: apps/mo1-ios/ios/Sources/Mo1Core/Contracts/Notification.swift
// Notification — inbox shapes consumed by the app. Mirrors the consumed subset
// of packages/types/src/notification.ts. NotificationType is an open vocabulary.
import Foundation

public typealias NotificationType = String

public struct InboxItem: Codable, Equatable, Sendable, Identifiable {
    public var id: Ids.NotificationId
    public var type: NotificationType
    public var title: String
    public var body: String
    public var readAt: ISODateTime?
    public var createdAt: ISODateTime
    public var resourceType: String?
    public var resourceId: String?
    public init(
        id: Ids.NotificationId, type: NotificationType, title: String, body: String,
        readAt: ISODateTime?, createdAt: ISODateTime, resourceType: String?, resourceId: String?
    ) {
        self.id = id; self.type = type; self.title = title; self.body = body
        self.readAt = readAt; self.createdAt = createdAt
        self.resourceType = resourceType; self.resourceId = resourceId
    }
}

public struct ListInboxQuery: Equatable, Sendable {
    public var cursor: String?
    public var limit: Int?
    public var unreadOnly: Bool?
    public init(cursor: String? = nil, limit: Int? = nil, unreadOnly: Bool? = nil) {
        self.cursor = cursor; self.limit = limit; self.unreadOnly = unreadOnly
    }
}

public typealias ListInboxResponse = Paginated<InboxItem>
